### PR TITLE
drt: Consolidate spacing calculation code

### PR DESCRIPTION
**Pull request by antonblanchard**
Signed-off-by: Anton Blanchard <anton@linux.ibm.com>

### CODE REVIEW

**Comment by antonblanchard** [comment]
There are quite a few more places to consolidate, but this was a simple subset to start with.

There are one or two places that error (or warn) when the layer doesn't have min spacing constraints, or if the constraint wasn't one of the types we expect. It seems like we could check this once during initialisation if it is required (these cases will now just use a space of 0).

File: src/drt/src/dr/FlexDR.cpp [hunk 1]
```diff
@@ -963,21 +926,8 @@ frCoord FlexDR::init_via2viaMinLenNew_minSpc(frLayerNum lNum,
...
-      reqDist = static_cast<frSpacingTablePrlConstraint*>(con)->find(
-          max(width1, width2), prl1);
-    } else if (con->typeId()
-               == frConstraintTypeEnum::frcSpacingTableTwConstraint) {
-      reqDist = static_cast<frSpacingTableTwConstraint*>(con)->find(
-          width1, width2, prl1);
-    }
-  }
+  reqDist = getTech()->getLayer(lNum)->getMinSpacingValue(
+      width1, width2, min(prl1, prl2), false);
```
Comment (maliberty) [inline src/drt/src/dr/FlexDR.cpp:930]: min(prl1, prl2) looks to be a mistranslation - should be prl1

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/drt/src/dr/FlexDR.cpp (hunk 1 above)
Comment (antonblanchard) [reply src/drt/src/dr/FlexDR.cpp]: Good catch, fixed

File: src/drt/src/db/tech/frLayer.cc [hunk 2]
```diff
@@ -125,3 +125,38 @@ void frLayer::printAllConstraints(utl::Logger* logger)
     }
   }
 }
+
+frCoord frLayer::getMinSpacingValue(frCoord width1,
+                                    frCoord width2,
+                                    frCoord prl,
+                                    bool use_min_spacing)
```
[the remark this answers is not in the document]
Comment (antonblanchard) [reply src/drt/src/db/tech/frLayer.cc:132]: There are a few cases that will use this, eg:

https://github.com/The-OpenROAD-Project/OpenROAD/blob/5addd6be325e7cf8aeffe522a4b077f1257581d4/src/drt/src/dr/FlexDR_maze.cpp#L296-L313

[the remark this answers is not in the document]
Comment (maliberty) [reply src/drt/src/db/tech/frLayer.cc:132]: So it is for future use.

**Review by maliberty — APPROVED** [review]

**Pull request merged**